### PR TITLE
Fix RPC's success code in case of UNSUPPORTED_RESOURCE

### PR DESCRIPTION
**Pull request by AKalinich-Luxoft**
Fixed Alert, AlertManeuver, PerformAudioPassThru success code in case of `UNSUPPORTED_RESOURCE`. The root cause of problem here is that SDL does not check specific case for this result code for RPC with two or more interfaces and just return wrong success result code.

In this pull request:
- Added overriden functions `PrepareResultForMobileResponse/PrepareResultCodeForResponse`
to cover specific cases for listed RPCs
- Updated unit test expectations for `PerformAudioPassThruRequestTest`

**Note**
Some changes are from PR #1567 and will disappear after rebase.

### CODE REVIEW

**Comment by AKalinich-Luxoft** [comment]
@AByzhynar, @LitvinenkoIra, @LuxoftAKutsan, @dev-gh, @anosach-luxoft please review

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h [hunk 1]
```diff
@@ -233,8 +260,8 @@ class CommandRequestImpl : public CommandImpl,
    * @return true if result code complies successful result code
    * otherwise returns false
    */
-  bool PrepareResultForMobileResponse(ResponseInfo& out_first,
-                                      ResponseInfo& out_second) const;
+  virtual bool PrepareResultForMobileResponse(ResponseInfo& out_first,
```
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @AKalinich-Luxoft Please rename parameters like you did in `IsResultCodeUnsupported(const ResponseInfo& first_iface_response...`

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h [hunk 2]
```diff
@@ -256,7 +283,7 @@ class CommandRequestImpl : public CommandImpl,
    * interface that returns response.
    * @return resulting code for sending to mobile application.
    */
-  mobile_apis::Result::eType PrepareResultCodeForResponse(
+  virtual mobile_apis::Result::eType PrepareResultCodeForResponse(
```
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @AKalinich-Luxoft Please rename parameters like you did in `IsResultCodeUnsupported(const ResponseInfo& first_iface_response...`

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h (hunk 1 above)
Comment (AKalinich-Luxoft) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @AByzhynar done in https://github.com/smartdevicelink/sdl_core/pull/1575/commits/596d8b9662b5da0b4c97f5d377d27bdfc0dfe16a

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h (hunk 2 above)
Comment (AKalinich-Luxoft) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @AByzhynar done in https://github.com/smartdevicelink/sdl_core/pull/1575/commits/596d8b9662b5da0b4c97f5d377d27bdfc0dfe16a

Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @AKalinich-Luxoft No requested changes provided

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h (hunk 1 above)
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @AKalinich-Luxoft No requested changes provided

Comment (AKalinich-Luxoft) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @AByzhynar sorry, wrong commit number. Please see 9e2ce859407b0ff871832e4f6e99254cc2eb4a8f

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h (hunk 2 above)
Comment (AKalinich-Luxoft) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @AByzhynar sorry, wrong commit number. Please see 9e2ce859407b0ff871832e4f6e99254cc2eb4a8f

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h [hunk 3]
```diff
@@ -226,15 +253,16 @@ class CommandRequestImpl : public CommandImpl,
...
+   * and interface that returns response
+   * @param second_iface_response contains result_code from second HMI response
+   * and interface that returns response
    * @return true if result code complies successful result code
    * otherwise returns false
    */
-  bool PrepareResultForMobileResponse(ResponseInfo& out_first,
-                                      ResponseInfo& out_second) const;
+  virtual bool PrepareResultForMobileResponse(
+      ResponseInfo& first_iface_response,
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @LuxoftAKutsan done in 28af6b88a83cb1dbe3421b446cd768c6b2e68a7e

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 4]
```diff
@@ -109,13 +109,23 @@ bool CheckResultCode(const ResponseInfo& first, const ResponseInfo& second) {
...
-bool IsResultCodeUnsupported(const ResponseInfo& first,
-                             const ResponseInfo& second) {
-  return ((first.is_ok || first.is_invalid_enum) &&
-          second.is_unsupported_resource) ||
-         ((second.is_ok || second.is_invalid_enum) &&
-          first.is_unsupported_resource) ||
-         (first.is_unsupported_resource && second.is_unsupported_resource);
+bool IsResultCodeUnsupported(const ResponseInfo& first_iface_response,
+                             const ResponseInfo& second_iface_response) {
+  return ((first_iface_response.is_ok ||
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @LuxoftAKutsan done in 28af6b88a83cb1dbe3421b446cd768c6b2e68a7e

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 5]
```diff
@@ -689,49 +699,57 @@ bool CommandRequestImpl::PrepareResultForMobileResponse(
...
 
-  out_first.is_invalid_enum =
-      hmi_apis::Common_Result::INVALID_ENUM == out_first.result_code;
+  first_iface_response.is_invalid_enum =
+      hmi_apis::Common_Result::INVALID_ENUM == first_iface_response.result_code;
 
-  out_second.is_invalid_enum =
-      hmi_apis::Common_Result::INVALID_ENUM == out_second.result_code;
+  second_iface_response.is_invalid_enum =
+      hmi_apis::Common_Result::INVALID_ENUM ==
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @LuxoftAKutsan yes, this code was formatted with clang-format-3.6

**Comment by AKalinich-Luxoft** [comment]
@LuxoftAKutsan please finish you review

**Pull request closed without merging**